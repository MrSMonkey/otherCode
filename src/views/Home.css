@import '../styles/variables.global.css';

.banner {
  position: relative;
  width: 100%;
  height: 398px;
  color: $color-white;
  p {
    position: absolute;
    left: 59px;
    letter-spacing: 1px; /* no */
  }
}

.asset-title {
  top: 100px;
  font-size: 12px; /* no */
  font-weight: lighter;
}

.asset-num {
  top: 133px;
  font-size: 32px; /* no */
}

.house-title {
  top: 261px;
  font-size: 12px; /* no */
  font-weight: lighter;
}

.house-num {
  top: 294px;
  font-size: 12px; /* no */
  font-weight: lighter;
}

.star-home {
  padding: 20px 30px 0 30px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  & img {
    margin-top: 20px;
    width: 335px;
    height: 200px;
    vertical-align: bottom;
  }
}

.call-number {
  color: $color-golden;
  font-size: 15px; /* no */
  line-height: 88px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  img {
    width: 32px;
    height: 41px;
  }
}

.hr-title {
  margin-top: 60px;
}

.smart-lock {
  position: relative;
  margin-top: 30px;
  padding-top: 102px;
  width: 100%;
  height: 1190px;
  color: $color-white;
  text-align: center;
  h3 {
    font-size: 28px; /* no */
    line-height: 60px;
    background-image: -webkit-gradient(
      linear,
      0 0,
      0 bottom,
      from(rgba(255, 255, 255, 1)),
      to(rgba(255, 255, 255, 0.2))
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
  button {
    width: 200px;
    height: 60px;
    color: #b8b8b8;
    background-color: transparent;
    border: 1px solid; /* no */
    font-size: 14px; /* no */
    font-weight: ;
  }
  button:focus {
    outline: 0;
  }
}

.lock-dec {
  margin-top: 40px;
  color: $color-gray-light;
  font-size: 12px; /* no */
  font-weight: lighter;
  span {
    color: $color-white;
  }
}

.lock-price {
  font-size: 32px; /* no */
  line-height: 90px;
  i {
    position: relative;
    left: 4px; /* no */
    font-style: normal;
  }
  span {
    font-size: 12px; /* no */
    color: $color-gray-light;
    text-decoration: line-through;
  }
}

.lock-footer {
  position: absolute;
  left: 50%;
  bottom: 45px;
  width: 100%;
  font-size: 10px; /* no */
  color: $font-color-light;
  transform: translateX(-50%);
}

.user-info {
  position: absolute;
  right: 40px;
  bottom: 65px;
  white-space: nowrap;
}
.user-avatar {
  display: inline-block;
  vertical-align: middle;
  width: 76px;
  height: 76px;
  border-radius: 50%;
  overflow: hidden;
  background-size: contain;
}

.user-nickname {
  display: inline-block;
  vertical-align: middle;
  max-width: 5em;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-left: 12px;
}
