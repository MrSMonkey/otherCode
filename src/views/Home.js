import React, { Component } from 'react'
import { connect } from 'react-redux'
import CSSModules from 'react-css-modules'
import { actions as myHouseActions } from '../reducers/myHouse'
import styles from './Home.css'
import HeaderBanner from '../components/HeaderBanner'
import HrTitle from '../components/HrTitle'
import UokoDetailPlant from '../components/UokoDetailPlant'
import Icon from '../components/Icon'
import { SMART_LOCK_LINK } from '../utils/const'

const imgs = {
  homeBg: require('../assets/imgs/home_bg.jpg'),
  lock: require('../assets/imgs/smart_lock.jpg'),
  xingkong: require('../assets/imgs/starhome_xingkong.png'),
  jiameng: require('../assets/imgs/starhome_jiameng.png'),
  youke: require('../assets/imgs/starhome_youke.png'),
  userAvatarDefault: require('../assets/imgs/profileimg.png'),
  yezhu: require('../assets/imgs/starhome_yezhu.png')
}

/**头图 */

const Banner = ({ data, loading, user }) => {
  return (
    <HeaderBanner img={imgs.homeBg}>
      <header styleName="banner">
      </header>
    </HeaderBanner>
  )
}
const StyleBanner = CSSModules(Banner, styles)

/**品牌群 */
const Products = props => {
  const starHomesImg = [
    {
      alt: '星空',
      src: imgs.xingkong
    },
    {
      alt: '星空家盟',
      src: imgs.jiameng
    },
    {
      alt: '星空有客',
      src: imgs.youke
    },
    {
      alt: '星空业主',
      src: imgs.yezhu
    }
  ]
  return (
    <div styleName="star-home">
      {starHomesImg.map((item, index) => {
        return <img src={item.src} alt={item.alt} key={index} />
      })}
    </div>
  )
}
const StyleProducts = CSSModules(Products, styles)

/**智能硬件 */
const SmartLock = () => {
  return (
    <div
      styleName="smart-lock"
      className="bg-img-full"
      style={{ backgroundImage: `url(${imgs.lock})` }}
    >
      <h3>UOKO | 智能门锁</h3>
      <p styleName="lock-dec">
        星空业主尊享
        <span>200元</span>
        优惠
      </p>
      <p styleName="lock-price">
        <i>￥</i>
        1299
        <span>￥1499</span>
      </p>
      <a href={SMART_LOCK_LINK}>
        <button type="button">优惠购买</button>
      </a>
      <p styleName="lock-footer">
        高端精致 | 性能卓越 | 工艺精良 | 安全可靠 | 实用主义
      </p>
    </div>
  )
}
const StyleSmartLock = CSSModules(SmartLock, styles)

@CSSModules(styles)
class Home extends Component {
  render() {
    const { income, incomeLoading, user } = this.props
    return (
      <div>
        <StyleBanner user={user} data={income} loading={incomeLoading} />
        <div styleName="hr-title">
          <HrTitle title="uoko星空产品服务群" data={income} />
        </div>
        <StyleProducts />
        <div styleName="hr-title">
          <HrTitle title="优客逸家" dec="连续4年最具影响力品牌" />
        </div>
        <UokoDetailPlant>
          <a href="tel:10105288">
            <div styleName="call-number">
              <span>1010-5288</span>
              <Icon name="call" width={32} height={41} />
            </div>
          </a>
        </UokoDetailPlant>
        <div styleName="hr-title">
          <HrTitle title="智能硬件" />
        </div>
        <StyleSmartLock />
      </div>
    )
  }
}

export default connect(
  state => ({
    income: state.myHouse.income,
    incomeLoading: state.myHouse.incomeLoading,
    user: state.app.landlordInfo
  }),
  myHouseActions
)(Home)
