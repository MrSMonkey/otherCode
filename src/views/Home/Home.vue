<!--
  /*
  * @Description: 首页
  * @Author: chenmo
  * @Date: 2018-09-17 17:20:01
 * @Last Modified by: chenmo
 * @Last Modified time: 2019-04-11 14:47:15
  */
-->

<template>
  <section class="home">
    <main class="home-comp_list">
      <section class="header-img">
        <img src="@/assets/images/home_bg.png" alt=""/>
      </section>
      <hr-title title="星空产品服务群"></hr-title>
      <section class="star-home">
        <img v-for="img in startHomeImg" :key="img.alt" alt="img.alt" :src="img.src"/>
      </section>
      <hr-title title="优客逸家" dec="连续四年最具影响力品牌"></hr-title>
      <UokoDetailPlant>
        <a href="tel:10105288">
          <div class="call-number">
            <span>1010-5288</span>
            <img  src="@/assets/images/icon/icon_call.png" alt="电话"/>
          </div>
        </a>
      </UokoDetailPlant>
      <hr-title title="智能硬件"></hr-title>
      <section class="smart-lock">
          <h3>UOKO | 智能门锁</h3>
          <p class="lock-dec">
            星空业主尊享
            <span>200元</span>
            优惠
          </p>
          <p class="lock-price">
            <i>￥</i>
            1299
            <span>￥1499</span>
          </p>
          <a :href="link">
            <button type="button">优惠购买</button>
          </a>
          <p class="lock-footer">
            高端精致 | 性能卓越 | 工艺精良 | 安全可靠 | 实用主义
          </p>
      </section>
    </main>
    <van-dialog
      v-model="show"
      title="版本更新记录"
      class="dialog"
    >
    <time-line :timeLineData="timeLineData"></time-line>
    </van-dialog>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Icon, Dialog } from 'vant';
import { State, Getter, Mutation, Action } from 'vuex-class';
import CommonMixins from '@/utils/mixins/commonMixins';
import HrTitle from '@/components/HrTitle.vue';
import UokoDetailPlant from '@/components/UokoDetailPlant.vue';
import TimeLine from '@/components/TimeLine.vue';
import timeLineData from '../../../data/versions';
import {START_HOME_IMG, SMART_LOCK_LINK} from '@/config/config';

// 声明引入的组件
@Component({
  name: 'Home',
  components: {
    [Icon.name]: Icon,
    HrTitle,
    UokoDetailPlant,
    TimeLine
  }
})
// 类方式声明当前组件
export default class Home extends CommonMixins {
  private startHomeImg: any[] = START_HOME_IMG;
  private link: string = SMART_LOCK_LINK;
  private show: boolean = false;
  private timeLineData: any = timeLineData;
  @State('version') private version: string;

}
</script>

<style lang="stylus" rel="stylesheet/stylus">
@import '../../assets/stylus/main.styl'

.home
  .header-img
    img 
      display inline-block
      width 100%
  .star-home
    padding vw(20)
    display flex
    flex-wrap wrap
    justify-content space-between
    img 
      margin-bottom 10px
      width vw(162)
      height 100px
      vertical-align bottom
  .smart-lock
    background-image url('../../assets/images/ad_bg.png');
    background-repeat no-repeat
    background-size cover
    background-position center
    width 100%
    height 595px
    margin-top vw(20)
    color #fff
    text-align: center;
    position relative
    h3
      font-size 28px /* no */
      line-height 60px
      background-image -webkit-gradient(
        linear,
        0 0,
        0 bottom,
        from(rgba(255, 255, 255, 1)),
        to(rgba(255, 255, 255, 0.2))
      )
      -webkit-background-clip text
      -webkit-text-fill-color transparent
    button
      width 100px
      height 30px
      margin vw(10) 0
      color #b8b8b8
      background-color transparent
      border 1px solid /* no */
      font-size 14px /* no */
    .lock-dec
      margin-top 10px
      color $color-gray-light
      font-size 12px /* no */
      font-weight lighter
      span
        color #fff
    .lock-price
      font-size 30px /* no */
      i
        position relative
        left 4px /* no */
        font-style normal
      span
        font-size 12px /* no */
        color $color-gray-light
        text-decoration line-through
    .lock-footer
      position: absolute
      left 50%
      bottom vw(10)
      width 100%
      font-size 10px /* no */
      color $next-text-color
      transform translateX(-50%)
  .dialog
    .van-dialog__header
      padding-top 15px !important
    .van-dialog__confirm, .van-dialog__confirm:active
      color $main-color
</style>
