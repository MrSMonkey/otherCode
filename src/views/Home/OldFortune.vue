/*
 * @Description: 星空财神
 * @Author: LongWei
 * @Date: 2019-04-23 15:46:54
 * @Last Modified by: LongWei
 * @Last Modified time: 2019-04-24 14:03:38
 */

<template>
  <section>
    <main class="fortune">
      <div class="god-pic-box">
        <img src="@/assets/images/god.png" alt="星空财神" />
        <div>
          <p>每天早餐该花多少钱？</p>
          <p>每月收入该怎么分配？</p>
        </div>
      </div>
      <p class="words">
        <i class="coin" />
        <span class="text">财富自由第一定律 财神星马上告诉您</span>
        <i class="coin" />
      </p>
      <div class="income-input-box">
        <van-field
          class="income-input"
          v-model="incomeValue"
          type="number"
          placeholder="老板，请填写您的月可支配现金收入" 
          clearable
        />

        <a @click="calcResult"  class="result-btn">
          请财神星掐指一算
        </a>
        </div>
        <div :class="isStopped ? 'anim-box' : 'anim-box show'">
          <Lottie :options="defaultOptions" v-on:animCreated="handleAnimation"  />
        </div>
    </main>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import CommonMixins from '@/utils/mixins/commonMixins';
import { Field, Notify } from 'vant';
import Lottie from 'vue-lottie';
import getWXConfig from '@/config/wxConfig';
import wxShareConfig from '@/config/wxShareConfig';
import * as Anim from '../../assets/json/fortune_anim.json';
import { SHARE_API_LIST, FORTUNE_SHARE_DATA } from '@/config/config';

// 声明引入的组件
@Component({
  name: 'Fortune',
  components: {
    [Field.name]: Field,
    [Notify.name]: Notify,
    Lottie,
  }
})
// 类方式声明当前组件
export default class Fortune extends CommonMixins {
  private incomeValue: string = '';
  private isStopped: boolean = true;
  private defaultOptions: any = { animationData: Anim.default, loop: false, autoplay: false};
  private animationSpeed: number = 1;
  private anim: any = {};
  private shareList: any[] = SHARE_API_LIST;
  private shareData: any = FORTUNE_SHARE_DATA;
  private mounted() {
    this.anim.addEventListener('complete', () => {
      this.anim.stop();
      this.$router.push({
          path: '/oldFortuneResult',
          query: {
            money: this.incomeValue,
          }
        });
    });

    getWXConfig(this.shareList).then((res: any) => {
      wxShareConfig(this.shareData);
    });
  }

  // 是否是正整数
  private isInteger(num: string) {
    if (num !== '0' && !num) {
      return false;
    }
    const numberValue: number = Number(num);
    if (isNaN(numberValue)) {
      return false;
    }
    if (numberValue <= 0) {
      return false;
    }
    return true;
  }

  // 计算
  private calcResult() {
    if (!this.isInteger(this.incomeValue)) {
      Notify('请输入大于0的数字!');
    }
    this.isStopped = false;
    this.anim.play();
  }

  // 计算
  private handleAnimation(anim: any) {
    this.anim = anim;
  }
}
</script>

<style lang="stylus" rel="stylesheet/stylus">

.fortune
  .god-pic-box
    height vw(325)
    line-height 0
    position relative
    img 
      width 100%
    div
      position absolute
      top vw(17)
      left 0
      width 100%
      text-align center
      line-height 140%
      color #A58D68
      font-size 18px /* no */
      font-weight 600
  .words 
    font-size 16px /* no */
    color #c3aa82
    text-align center
    font-weight bold
    margin-top 25px
    margin-bottom 30px
    .coin 
      width vw(17)
      height vw(17)
      display inline-block
      line-height 1
      background-image url('../../assets/images/coin.png')
      background-repeat no-repeat;
      background-size contain
      vertical-align middle
    .text 
      margin 0 vw(5)
      display inline-block
      vertical-align middle
  .income-input-box 
    padding: 0 vw(27.5);
  .income-input 
    padding vw(10) vw(5)
    font-size:15px /* no */
    border 1px solid #d8d8d8 /* no */
    border-radius 8px
    -webkit-appearance none
    display block
    width 100%
    box-sizing border-box
    outline none
  .result-btn 
    display block
    width 100%
    height 44px
    line-height 44px
    font-size 0
    background-image linear-gradient(-270deg, #C3AA82 0%, #E2CBA7 100%)
    border-radius 8px
    overflow hidden
    text-align center
    margin-top vw(10)
    color #fff
    font-size 15px /* no */
    transition transform 0.05s ease
    &:active 
      transform: scale(0.98);
  .anim-box 
    height: 100%
    position absolute
    left 0
    right 0
    bottom 0
    top 0
    background: rgba(0, 0, 0, 0.7) url('../../assets/images/bg_light.png')  center/cover
    display: none
    &.show 
      display block
</style>
