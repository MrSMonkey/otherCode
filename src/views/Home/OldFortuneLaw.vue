/*
 * @Description: 星空财神 - 4321定律
 * @Author: LongWei
 * @Date: 2019-04-23 15:46:54
 * @Last Modified by: LongWei
 * @Last Modified time: 2019-05-07 17:16:13
 */

<template>
  <section>
    <main class="law">
      <section class="law-header">
        <p>您知道</p>
        <p>巴菲特的早餐费，由什么决定吗？</p>
        <p>是的，由股票涨跌决定的~</p>
        <p>涨的时候吃3.27美元的早餐，跌的时候吃2.21美元的套餐</p>
        <p>巴菲特~美国最富有的人，是按照什么逻辑在安排这一切？</p>
        <p>这就是资产配置的核心-----比例投资法</p>
        <p>4321定律是其中最方便实用的收入分配法</p>
      </section>
      <section>
        <h2 class="tit">
          <i class="coin" />
          <span class="tit-text">举例：月可支配收入为10000元</span>
        </h2>
        <p>
          4000元：10000*
          <span class="text-spec">40%投资理财</span>
          ：比如投资股票、外汇、基金等有较高收益率的资产，也可以选择开放式基金定期定额投资。
        </p>
        <p>
          3000元：10000* <span class="text-spec">30%衣食住行</span>
          ：每月基本不可缺少的生活费用。吃饭穿衣费、手机费、交通、房租或按揭等。
        </p>
        <p>
          2000元: 10000*
          <span class="text-spec">20%储蓄备用</span>
          ：存为活期存款，在需要的时候可以方便的提出来，用于改善生活质量。相当于家庭备用金。
        </p>
        <p>
          1000元: 10000*
          <span class="text-spec">10%保险</span>
          ：投保是一种长远的安排，也是最容易忽略的投资方式，是对日后生活的负责和保障，尤其是预防家庭收入的主要创造者可能遇到的意外情况，以免对家庭经济造成重创。以保额（即出险后保险公司的赔付额）一般不低于年收入的10倍为合适。
        </p>
      </section>
    </main>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import CommonMixins from '@/utils/mixins/commonMixins';


// 声明引入的组件
@Component({
  name: 'OldFortuneLaw',
})
// 类方式声明当前组件
export default class OldFortuneLaw extends CommonMixins {
  private money: string = '';
}
</script>

<style lang="stylus" rel="stylesheet/stylus">

.law
  background-color $bg-color-default
  padding vw(15) vw(20)
  .tit 
    font-weight bold
    font-size 14px /* no */
    margin-top vw(25)
    margin-bottom vw(10)
  .coin 
    width vw(17)
    height vw(17)
    display inline-block
    line-height: 1
    background-image url('../../assets/images/coin.png')
    background-repeat no-repeat
    background-size contain
    vertical-align middle
    margin-right vw(6)
  .tit-text 
    display inline-block
    vertical-align middle
  .text-spec 
    color $main-color
</style>
