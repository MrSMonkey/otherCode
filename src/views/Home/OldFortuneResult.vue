/*
 * @Description: 星空财神 - 4321定律中间页
 * @Author: LongWei
 * @Date: 2019-04-23 15:46:54
 * @Last Modified by: LongWei
 * @Last Modified time: 2019-04-24 14:05:07
 */

<template>
  <section>
    <main class="result-box">
      <div class="title">
        <h2>【4321定律】</h2>
        <h2>财富自由第一定律</h2>
      </div>
      <div class="result-cont">
        <div class="p1">
          <p class="p-tit">投资理财</p>
          <p class="p-money">
            <span class="p-num">{{ getIncomePart(money, 0.4)}}</span>元
          </p>
          <p class="p-desc">收入之40%用于投资理财、财富增值</p>
        </div>
        <div class="p2">
          <p class="p-tit">家庭开支</p>
          <p class="p-money">
            <span class="p-num">{{ getIncomePart(money, 0.3)}}</span>元
          </p>
          <p class="p-desc">收入之30%用于家庭开支</p>
        </div>
        <div class="p3">
          <p class="p-tit">活期存款</p>
          <p class="p-money">
            <span class="p-num">{{ getIncomePart(money, 0.2)}}</span>元
          </p>
          <p class="p-desc">收入之20%用于活期存款</p>
        </div>
        <div class="p4">
          <p class="p-tit">保险投资</p>
          <p class="p-money">
            <span class="p-num">{{ getIncomePart(money, 0.1)}}</span>元
          </p>
          <p class="p-desc">收入之10%用于个人/家庭保险</p>
        </div>
        <router-link to="/oldFortuneLaw" class="p-center">
          4321定律
        </router-link>
      </div>
    </main>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import CommonMixins from '@/utils/mixins/commonMixins';


// 声明引入的组件
@Component({
  name: 'FortuneResult',
})
// 类方式声明当前组件
export default class FortuneResult extends CommonMixins {
  private money: string = '';

  private mounted() {
    this.money = String(this.$route.query.money);
  }

  // 计算百分比
  private getIncomePart(total: string, percent: number) {
    let totalNum: number = Number(total);
    if (isNaN(totalNum)) {
      totalNum = 0;
    }
    return Math.floor(totalNum * percent);
  }
}
</script>

<style lang="stylus" rel="stylesheet/stylus">

.result-box
  padding-top vw(20)
  background: $bg-color-default;
  .title 
    h2 
      font-size 24px /* no */
      text-align center
      color #b89e75
  .result-cont 
    margin 0 auto
    padding vw(20)
    overflow hidden
    position relative
    .p1,.p2,.p3,.p4 
      width vw(162.5)
      height vw(210)
      background-size cover
      background-repeat no-repeat
      color #fff
      float left
      margin-bottom vw(10)
      box-sizing border-box
      padding vw(15)
      &:nth-of-type(2n+1)
        margin-right vw(10)
      &:nth-of-type(2n+2)
        text-align right
    .p1
      background-image url('../../assets/images/p1.png')
    .p2 
      background-image url('../../assets/images/p2.png')
    .p3
      background-image url('../../assets/images/p3.png')
    .p4
      background-image url('../../assets/images/p4.png')
    .p-tit 
      font-size 18px /* no */
    .p-money 
      font-size 18px/* no */
      margin-bottom vw(14)
      margin-top vw(3.5)
    .p-num 
      font-weight bold
      font-size 30px /* no */
    .p-desc 
      font-size 14px /* no */
      text-align left
    .p-center 
      position absolute
      width vw(116.5)
      height vw(116.5)
      background-image url('../../assets/images/btn_center.png')
      background-size contain
      overflow hidden
      left 0
      top vw(-18)
      right 0
      bottom 0
      margin auto
      font-size 0
      &:active
        transform  scale(0.98)
</style>
