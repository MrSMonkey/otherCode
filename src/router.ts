import Vue from 'vue';
import Router from 'vue-router';
import Home from '@/views/Home/Home.vue';
Vue.use(Router);

let router!: any;

const Bind = () => import(/* webpackChunkName: 'bind' */ '@/views/Login/Bind.vue');

const Entrust = () => import(/* webpackChunkName: 'entrust' */ '@/views/House/Entrust.vue');
const Perfect = () => import(/* webpackChunkName: 'perfect' */ '@/views/House/Perfect.vue');
const House = () => import(/* webpackChunkName: 'house' */ '@/views/House/House.vue');
const EntrustPlan = () => import(/* webpackChunkName: 'entrustPlan' */ '@/views/House/EntrustPlan.vue');
const ChoiceHouse = () => import(/* webpackChunkName: 'choiceHouse' */ '@/views/House/ChoiceHouse.vue');

const MyHouse = () => import(/* webpackChunkName: 'myHouse' */ '@/views/HouseDetail/MyHouse.vue');
const HouseImages = () => import(/* webpackChunkName: 'houseImages' */ '@/views/HouseDetail/HouseImages.vue');

const ServiceOrder = () => import(/* webpackChunkName: 'serviceOrder' */ '@/views/ServiceOrder/ServiceOrder.vue');
const ServiceDetile = () => import(/* webpackChunkName: 'serviceDetile' */ '@/views/ServiceOrder/ServiceDetile.vue');
const ProductDetile = () => import(/* webpackChunkName: 'productDetile' */ '@/views/ServiceOrder/ProductDetile.vue');
const ServiceOrderDetail = () => import(/* webpackChunkName: 'serviceOrderDetail' */ '@/views/ServiceOrder/ServiceOrderDetail.vue');
const ServiceInfo = () => import(/* webpackChunkName: 'serviceInfo' */ '@/views/ServiceOrder/ServiceInfo.vue');
const ProductInfo = () => import(/* webpackChunkName: 'productInfo' */ '@/views/ServiceOrder/ProductInfo.vue');

const Purchase = () => import(/* webpackChunkName: 'purchase' */ '@/views/Service/Purchase.vue');
const ServiceType = () => import(/* webpackChunkName: 'serviceType' */ '@/views/Service/ServiceType.vue');
const StartService = () => import(/* webpackChunkName: 'startService' */ '@/views/Service/StartService.vue');
const NotFoundComp = () => import(/* webpackChunkName: '404' */ '@/components/404.vue');

const MaintainChecked = () => import(/* webpackChunkName: 'maintainChecked' */ '@/views/ServiceRecord/MaintainChecked.vue');
const ServiceRecord = () => import(/* webpackChunkName: 'serviceRecord' */ '@/views/ServiceRecord/ServiceRecord.vue');
const ConfirmPay = () => import(/* webpackChunkName: 'confirmPay' */ '@/views/ServiceRecord/ConfirmPay.vue');
const ServiceRecordLook = () => import(/* webpackChunkName: 'serviceRecordLook' */ '@/views/ServiceRecord/ServiceRecordLook.vue');

const HouseAppraise = () => import(/* webpackChunkName: 'houseAppraise' */ '@/views/HouseAppraise/HouseAppraise.vue');
const AppraiseHouseInfo = () => import(/* webpackChunkName: 'houseAppraise' */ '@/views/HouseAppraise/AppraiseHouseInfo.vue');

router = new Router({
  base: process.env.BASE_URL,
  routes: [
    { path: '/', name: 'index', component: Home, meta: '星空业主服务号' },
    { path: '/entrustPlan', name: 'entrustPlan', meta: '星级房屋托管计划', component: EntrustPlan},
    { path: '/bind', name: 'bind', meta: '登录', component: Bind},
    { path: '/entrust', name: 'entrust', meta: '在线委托', component: Entrust},
    { path: '/perfect', name: 'perfect', meta: '完善房源信息', component: Perfect},
    { path: '/house', name: 'house', meta: '我的房源列表', component: House},
    { path: '/choiceHouse', name: 'choiceHouse', meta: '选择房源', component: ChoiceHouse},
    { path: '/myHouse', name: 'myHouse', meta: '房源详情', component: MyHouse},
    { path: '/serviceOrder', name: 'serviceOrder', meta: '服务订单', component: ServiceOrder},
    { path: '/serviceDetile', name: 'serviceDetile', meta: '订单详情', component: ServiceDetile},
    { path: '/productDetile', name: 'productDetile', meta: '订单详情', component: ProductDetile},
    { path: '/purchase', name: 'purchase', meta: '购买服务包', component: Purchase},
    { path: '/serviceInfo', name: 'serviceInfo', meta: '服务包详情', component: ServiceInfo},
    { path: '/productInfo', name: 'productInfo', meta: '服务产品详情', component: ProductInfo},
    { path: '/serviceType', name: 'serviceType', meta: '选择服务类型', component: ServiceType},
    { path: '/startService', name: 'startService', meta: '发起服务', component: StartService},
    { path: '/404', name: '404', meta: '404', component: NotFoundComp},
    { path: '/maintainChecked', name: 'maintainChecked', meta: '确认验收', component: MaintainChecked},
    { path: '/serviceRecord', name: 'serviceRecord', meta: '服务记录', component: ServiceRecord},
    { path: '/confirmPay', name: 'confirmPay', meta: '确认支付', component: ConfirmPay},
    { path: '/serviceRecordLook', name: 'serviceRecordLook', meta: '服务记录', component: ServiceRecordLook},
    { path: '/serviceOrderDetail', name: 'serviceOrderDetail', meta: '服务详情', component: ServiceOrderDetail},
    { path: '/serviceOrderDetail', name: 'serviceOrderDetail', meta: '服务详情', component: ServiceOrderDetail},
    { path: '/houseImages', name: 'houseImages', meta: '房源照片', component: HouseImages},
    {
      path: '/houseAppraise',
      name: 'houseAppraise',
      meta: '房屋估价',
      component: HouseAppraise,
      children: [
        {path: 'appraiseHouseInfo', name: 'appraiseHouseInfo', meta: '估价房屋信息', component: AppraiseHouseInfo }
      ]
    }
  ]
});

router.beforeEach((to: any, from: any, next: any) => {
  /* 路由发生变化修改页面title */
  if (to.meta) {
    document.title = to.meta;
  }

  // 未匹配到任何页面, 跳转到404页面
  if (to.matched.length === 0) {
    router.push('/404');
  }
  if (to.path === '/404') {
    next();
    return;
  }

  const token: any = localStorage.getItem('siteToken');
  if (!token) {
    // 非登陆状态
    if (to.path !== '/bind' && to.path !== '/entrust' && to.path !== '/entrustPlan' && to.path !== '/') {
      // 除了登录页 && 在线委托页 && 星级房屋托管计划，其他将跳转到登陆页
      router.push('/bind');
    }
    next();
  }
  next();
});

export default router;

