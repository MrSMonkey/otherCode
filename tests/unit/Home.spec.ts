/*
 * @Description: Home组件单元测试
 * @Author: LiuZhen
 * @Date: 2018-10-10 17:35:09
 * @Last Modified by: LiuZhen
 * @Last Modified time: 2019-01-22 14:57:25
 */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import router from 'vue-router';
import Vuex from 'vuex';
import Home from '@/views/Home.vue';

// 独立构建Vue构造函数
const localVue = createLocalVue();
localVue.use(router);
localVue.use(Vuex);

describe('对Home组件进行的单元测试', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        global: {}
      }
    });

    // 挂载组件
    wrapper = shallowMount(Home, {
      store,
      localVue
    });
  });

  afterEach(() => {
    // 销毁实例
    wrapper.destroy();
  });

  test('home', () => {
    // 测试实例是Vue组件
    expect(wrapper.isVueInstance()).toBeTruthy();
    // 存在 Home 组件
    expect(wrapper.is(Home)).toBeTruthy();
    // 测试 home 节点正确挂载
    expect(wrapper.find('.home').exists()).toBeTruthy();
  });

  // 生成测试快照
  test('create expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
