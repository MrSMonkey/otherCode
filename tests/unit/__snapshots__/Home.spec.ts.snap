// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`对Home组件进行的单元测试 create expected html structure 1`] = `
<section
  class="home"
>
  <header
    class="header"
  >
    <a
      class="aside-link"
      href="javascript:;"
    >
      <van-icon-stub
        class="aside-btn"
        classprefix="van-icon"
        name="wap-nav"
      />
    </a>
  </header>
   
  <section
    class="home-logo"
  >
    <van-row-stub
      class="logo-wrapper"
      gutter="0"
      tag="div"
    >
      <van-col-stub
        class="logo-img"
        span="9"
        tag="div"
      >
        <img
          alt=""
          src="[object Object]"
        />
      </van-col-stub>
       
      <van-col-stub
        span="15"
        tag="div"
      >
        UOKO Mobile Web
      </van-col-stub>
    </van-row-stub>
  </section>
   
  <main
    class="home-comp_list"
  >
    <router-link-stub
      event="click"
      tag="a"
      to="/businessOpa"
    >
      数据平台
    </router-link-stub>
  </main>
</section>
`;
